ShareX.HelpersLib/Helpers/ImageHelpers.cs

```diff
@@ -1596,6 +1664,91 @@ public static void FastBoxBlur(Bitmap bmp, int radius)
             }
         }
 
+        public static Bitmap WavyEdges(Bitmap bmp, int waveDepth, int waveRange, AnchorStyles sides)
```

Comment:
You should add this as image effect too, it is cool!

Reply:
By the way wouldn't this option also work?

![](https://jaex.getsharex.com/2022/08/ShareX_BkqqYvYM8V.png)

Reply:
![image](https://user-images.githubusercontent.com/1062071/185243088-94b7dacc-76e2-4124-a935-5df40b7f4720.png)

I do wonder if it would make sense to integrate some drop shadow or outline-drawing in the cut out effect, or at least a way to draw it on a non-transparent background. Right now it "lies" in the editor where it displays as blackness in the cut-out effect where it's actually transparent.

Reply:
I don't think we should add drop shadow.

Reply:
What you mean it lies and displays blackness? I literally draw black there in my screenshot myself to make it easily visible.

Reply:
I did try using the CurvedEdges flag for TornEdges before writing the WavyEdges function. It doesn't give the same kind of uniform result that WavyEdges does, where the two sides of the cut perfectly follow each other, so both have their uses depending on the "mood" the user wants to convey.

Reply:
> What you mean it lies and displays blackness? I literally draw black there in my screenshot myself to make it easily visible.

In the image editor it looks like this:
![image](https://user-images.githubusercontent.com/1062071/185346670-f7c13bdd-3112-4ece-b231-95c2082bd814.png)

Like the cut-out effect is a black river.
But when you save the image as PNG, or add effects to it that use transparency, it becomes clear that the "black river" is in fact a transparent area.

Reply:
Okay, when using a large Torn Edge effect that leaves "big lakes" instead of just a "narrow river", it becomes slightly visible that the black area is in fact a very dark checkerboard indicating transparency. That's very difficult to tell when it contrasts hard with the surrounding UI and/or the user's image being edited. I would suggest that the checkerboard effect used depends on the user's UI preference for light/dark.

Reply:
> I would suggest that the checkerboard effect used depends on the user's UI preference for light/dark.

It already does:

![](https://jaex.getsharex.com/2022/08/ShareX_SgOCU17Huu.png)

![](https://jaex.getsharex.com/2022/08/ShareX_LWGoTOEJLH.png)

Reply:
I have no idea what was wrong with my settings then, but toggling custom theme on and back off fixed it.